### PR TITLE
Resolving Issues 13, 14, 15

### LINKED ISSUE #13: Consolidate Allo.sol
per #11 

- merge concepts and relevant concepts _from_ IAllo.sol _into_ Allo.sol
- remove IAllo.sol

### LINKED ISSUE #14: Consolidate Registry.sol
per #11 

- merge concepts and comments _from_ IRegistry.sol _into_ Registry.sol
- Remove IRegistry.sol

### LINKED ISSUE #15: Reorganize file structure
Per #11, we want to reorganize the `contracts/` folder

The structure should be:
- interfaces/
    - IAllocationStrategy.sol
    - IDistributionStrategy.sol
- core/
    - libraries/Metadata.sol
    - Allo.sol
    - Registry.sol
    - Aqueduct.sol (note the slight spelling fix)
- strategies/
    - allocation/
    - distribution/

Note: this removes the `utils/` folder (which just duplicates the logic in core/libraries/Metadata.sol).

### PR DESCRIPTION
**Pull request by codenamejason**
### Description

- updates folder structure 
- remove interfaces
- removed unused contracts
- move MetaPtr from utils to core/library

Closes #13 
Closes #14
Closes #15

### CODE REVIEW

File: contracts/core/Aquaduct.sol
Comment (nfrgosselin) [inline contracts/core/Aquaduct.sol:1]: quick spelling fix —> this should be "Aqueduct.sol"

**Comment by zobront** [comment]
@codenamejason Looks like the file architecture is still a bit off:
- interfaces can move to the `contracts/` folder, outside of `core/`
- rename Aquaduct.sol to Aqueduct.sol
- add a `strategies/allocation` folder
- get rid of the `utils/` folder 

Can you also include all the comments from the IAllo and IRegistry contracts into the Allo.sol and Registry.sol contracts, above the correct functions (or inside those functions as comments)?

**Comment by zobront** [comment]
Looks good to me.

**Review by thelostone-mc — APPROVED** [review]
LGTM !

**Pull request merged**